### DISCUSSION TITLE
How to debug Bluepill Clones with ID 0x2ba01477

### DISCUSSION THREAD

**Discussion by danyhm** [stm32duino/Arduino_Core_STM32#2185] [Q&A] [answered]
Hello,

I've been able to debug an original Bluepill without problem, however, I also have a clone Bluepill with ID 0x2ba01477 and openocd is giving me the error 
```
Warn : UNEXPECTED idcode: 0x2ba01477
Error: expected 1 of 1: 0x1ba01477
```

is there any way to pass the correct parameters to openocd to debug this clone?
some people at Platformio have suggested some ways but they don't work for Arduino IDE
https://community.platformio.org/t/debugging-of-stm32f103-clone-bluepill-board-wrong-idcode/14635
https://community.platformio.org/t/error-while-trying-to-upload-code-to-bluepill-solved/22036

for example to pass .SVD files you create a debug_custom.json file in the sketch directory at pass it the SVD file like below
```
{
"svdFile": "STM32F103xx-CM3.svd"
}
```
is there any equvelant to pass "set CPUTAPID 0x2ba01477" ?

also for general knowledge my sketch runs fine on the original bluepill but gets into hardfault on the clone. so there is a hardware difference.

**Comment by fpistm** [comment, marked as answer]
Already answered:
https://github.com/orgs/stm32duino/discussions/2178#discussioncomment-7542007

Anyway you can try on the forum.
